src/utils/i18n.ts

```diff
@@ -334,3 +334,39 @@ export function lookupPhrase(options: LookupOptions): string {
   // Fallback to the key.
   return key;
 }
+
+/**
+ * Merges a phrase object into zero or more phrase objects.
+ *
+ * The context names of keys will also be validated.
+ */
+export function mergePhrases(targets: Phrases[], phrases: Phrases): void {
+  for (const context in phrases) {
+    if (!context.match(VALID_KEY_CONTEXT)) {
+      throw new Error(
+        `${context} is not a valid key context. ` +
+        `Valid keys should be of the format ${VALID_KEY_CONTEXT.toString()}.`);
+    }
+
+    const value = phrases[context];
+
+    if (isPhrases(value)) {
+      let newTargets: Phrases[] = [];
+      for (const target of targets) {
+        const contextedTarget = target[context];
+        if (!isPhrases(contextedTarget)) {
+          const newTarget: Phrases = {};
+          target[context] = newTarget;
+          newTargets.push(newTarget);
+        } else {
+          newTargets.push(contextedTarget);
+        }
+      }
+      mergePhrases(newTargets, value);
```

Comment:
There's also some awkwardness to avoid multiple calls to `mergePhrases` when given multiple `targets`/locales. It's probably more natural to put the if/else inside the loop rather than the other way around, but we want to avoid fanning out so that we don't have to validate the context more than we need to.

Reply:
I mention this elsewhere but this function is validating the input as it mutates the target, which means that if the function fails the target will be partially mutated. This __feels__ undesirable to me so perhaps the concerns of performance here are mute because it needs to iterate through the phrases two times?

---------

> There's also some awkwardness to avoid multiple calls to mergePhrases when given multiple targets/locales. It's probably more natural to put the if/else inside the loop rather than the other way around, but we want to avoid fanning out so that we don't have to validate the context more than we need to.

Yeah. It's probably worth leaving a comment about that. 

I might be bad at math, but in terms of asymptotic analysis whether you make the call inside of each loop or whether you merge for all the subtrees at the same time. The result is going to be either `(|target| * n) * |value|` or `|target| * (|value| * n)` regardless, where `n` is the number of targets. Due to the associative property the result is the same. This of course presumes that validating the key is constant time, which maybe it shouldn't be 🤷 .

Either way, I don't think this approach is bad, but just be sure to leave a comment:

```ts
// Recursively merge the phrases subtree for each target.
// Perform to avoid multiple validations.
```

Reply:
> I mention this elsewhere but this function is validating the input as it mutates the target, which means that if the function fails the target will be partially mutated.

That's a good point. I think doing the validation prior to merging would address this.

> I might be bad at math, but in terms of asymptotic analysis whether you make the call inside of each loop or whether you merge for all the subtrees at the same time.

I mistakenly assumed this'd be exponential but you're right. It would just do it twice for each key past the first depth. I'll be moving validation to a separate step so I guess this is moot anyway.